frontend/lib/Repository/EndpointRepository/EndpointCache.dart

```diff
@@ -0,0 +1,52 @@
+import 'package:flutter/material.dart';
+
+import '../../DataModels/Endpoint.dart';
+import '../../DataModels/EndpointData.dart';
+import '../../DataModels/EndpointSummary.dart';
+
+class EndpointCache {
+  final List<EndpointSummary> _endpointSummaryList = [];
+  final List<Endpoint> _endpointList = [];
```

Comment:
Może moglibyśmy w tym miejscu zamiast tych dwóch list użyć dwóch `Map`, gdzie kluczem byłoby `id`? Bo widzę, że robimy dużo operacji na szukanie elementu, a dzięki temu kod by się nam trochę uprościł i wydajnościowe też powinno być lepiej

Reply:
work in progress byku, ale wrzucaj co Ci do głowy przyjdzie